src/lane.hpp

```diff
@@ -2,25 +2,42 @@
 
 #include "common.hpp"
 #include "car.hpp"
+#include "direction.hpp"
+#include "car_type.hpp"
 
-class Lane : public Renderable
+#include <vector>
+
+class Lane
 {
 public:
-	const int MaxCarsPerLane = 5;
-
-	// Creates instance
-	bool init(float world_scale);
+	const int MaxCarsPerLane = 4;
 
-	// Releases instance
-	void destroy();
+    // constructor/destructor
+    Lane(float world_scale, direction dir);
+    ~Lane();
 
 	// Returns the current time remaining on the timer
     float get_time_remaining() const;
 
     // Returns the cars in the lane
-    Car* get_cars() const;
+    std::vector<Car> get_cars() const;
+
+    // Moves the game ahead by ms milliseconds
+    bool update(float ms);
+
+    // Adds the given car to the back of the lineup
+    void add_car(carType type);
+
+    // Removes the car at the front of the lane
+    void turn_car();

```

Comment:
I think advance_car or even move_car is better since many cars will go straight

Reply:
ehhh, those are kind of overloaded. I was thinking `advance` would be used to shift the cars forward in line after the first car has turned, and `move` is used in the Car class to actually move the sprite across the screen. `turn` would only be used for the car actually moving through the intersection, which I feel like is clear enough even if it doesn't actually involve a turn

Reply:
Fair enough. I personally found it confusing in the cpp file but I don't think it's a big deal!